sites/cheerpj/src/content/docs/13-tutorials/05-interoperability-tutorial.mdx

```diff
@@ -0,0 +1,185 @@
+
+### 1. Include the CheerpJ Loader Script
+
+Add the following script in the `<head>` section of `index.html`:
+
+```js title="index.html"
+<script src="https://cjrtnc.leaningtech.com/3.0/cj3loader.js"></script>
+```
+
+### 2. Define JavaScript Native Methods
+
+In the `<script>` tag, implement the native methods for interacting with Java.
+
+To implement the native methods in JavaScript, we create an `async` function for each Java method. You can move these methods to a separate `.js` file for better organization and include them using the `<script>` tag in the HTML.
+
+```js title="index.html"
+async function Java_com_example_Example_sendToHTML(lib, str) {
+	document.getElementById("javaOutput").innerText =
+		"JavaScript received: " + str;
+	console.log("Received input from Java: " + str);
+}
+
+async function Java_com_example_Example_nativeSetApplication(
+	lib,
+	myApplication
+) {
+	window.myApplication = myApplication;
+	console.log("Java application instance set on JavaScript side.");
+	return new Promise(() => {}); // Keeps the function from returning
+}
+```
+
+The `Java_com_example_Example_nativeSetApplication` function:
+
+- Assigns the Java application instance to `window.myApplication`.
+- Ensures continuous access by preventing the function from returning.
+
+The `Java_com_example_Example_sendToHTML` function:
+
+- Receives input from Java.
+- Displays the input in the HTML document.
+
+### 3. Initialize CheerpJ
+
+To load CheerpJ, we initialize it in the HTML file with specific configurations for native methods.
+
+```js title="index.html"
+(async () => {
+	await cheerpjInit({
+		version: 8,
+		natives: {
+			Java_com_example_Example_sendToHTML,
+			Java_com_example_Example_nativeSetApplication,
+		},
+	});
+	cheerpjCreateDisplay(800, 600);
+	await cheerpjRunJar("/app/example.jar");
+})();
+```
+
+Explanation:
+
+- `natives`: Lists JavaScript implementations of native Java methods. The naming convention follows `Java_<fully-qualified-class-name>_<method-name>`.
+- `cheerpjCreateDisplay`: Creates a display area for any graphical output (optional in this example).
+- `cheerpjRunJar`: Loads and runs the provided JAR file.
+
+### 4. Implement the `sendInputToJava` Function
+
+Write a function to send input to Java:
+
+```js title="index.html"
+async function sendInputToJava() {
+	const inputText = document.getElementById("inputText").value;
+	const response = await window.myApplication.processInput(inputText);
+}
+```
+
+The `sendInputToJava` function:
+
+- Retrieves input from an HTML input box.
+- Sends it to the Java method and displays the response in an alert.
+
+## Running the Example
+
+1. **Use the Provided `.jar` File**: If no code changes are made, you can directly use the precompiled `example.jar`.
+   **Recompile the `.jar` File**: If you modify the Java code, recompile it using the provided `Makefile`:
```

Comment:
We should point out that recompiling the java code requires a java installation. And also point out that we only changed java script code in the tutorial, so recompiling is not needed. If people want to play around more with the java code they can use the make file to recompile it.